transmission_interface/include/transmission_interface/transmission_parser.hpp

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TRANSMISSION_INTERFACE__TRANSMISSION_PARSER_HPP_
+#define TRANSMISSION_INTERFACE__TRANSMISSION_PARSER_HPP_
+
+#include <string>
+#include <vector>
+
+#include "transmission_interface/transmission_info.hpp"
+#include "transmission_interface/visibility_control.h"
+
+namespace transmission_interface
+{
+/**
+ * @brief Parse transmission information from a URDF
+ * @param urdf A string containing the URDF xml
+ * @return parsed transmission information
+ * @throws std::runtime_error on malformed or empty xml
+ */
```

Comment:
ROS2 uses Doxygen in the style of `///` and `/** */`

Reply:
I'm sorry I haven't yet seen an example :P

Reply:
I'm sorry but I don't think there's value in discussing doxygen styles as this is literally the first properly documented function in this repo.

For ref I wanted to look at something properly done but I found `/** */` and `///`  is only used for shorts as far as I can tell.

https://github.com/ros2/rclcpp/blob/master/rclcpp_components/include/rclcpp_components/node_instance_wrapper.hpp#L43
https://github.com/ros2/rclcpp/blob/master/rclcpp_components/include/rclcpp_components/component_manager.hpp

Also I'm sorry but I don't think there's value in discussing doxygen styles as this is literally the first documented function in this repo.

Reply:
c.f. https://github.com/ros2/rclcpp/blob/master/rclcpp_components/include/rclcpp_components/component_manager.hpp#L91-L99

Reply:
Fine, I've moved the brief into ///